client/state/posts/utils.js

```diff
@@ -208,3 +214,36 @@ export function getTermIdsFromEdits( post ) {
 		} )
 	};
 }
+
+/**
+ * Returns a normalized post terms object for sending to the API
+ *
+ * @param  {Object} post Raw post object
+ * @return {Object}      Normalized post object
+ */
+export function normalizeTermsForApi( post ) {
+	if ( ! post || ! post.terms ) {
+		return post;
+	}
+
+	return {
+		...post,
+		terms: pickBy( post.terms, ( terms ) => {
```

Comment:
I believe the server-side will merge terms and categories values, so it's not possible to override by passing an empty array, in which case we could do a filter instead of picking where assuming the first entry is an indicator that the rest are safe. What do you think?

```js
terms: mapValues( post.terms, ( terms ) => filter( terms, isString ) )
```

e.g.

```text
> mapValues( { category: [ { ID: 1 } ], post_tag: [ 'foo' ] }, ( terms ) => filter( terms, isString ) )
{ category: [], post_tag: [ 'foo' ] }
```